Framework/PythonInterface/plugins/algorithms/PEARLTransfit.py

```diff
@@ -0,0 +1,371 @@
+            #
+            # New Voigt function as from Igor pro function
+            Fit(Function='name=PEARLTransVoigt,Position=' + str(S_fit.column(1)[0]) + ',LorentzianFWHM='
+                         + str(lorentzFWHM) + ',GaussianFWHM=' + str(gaussianFWHM) + ',Amplitude='
+                         + str(S_fit.column(1)[3]) + ',Bg0=' + str(S_fit.column(1)[4]) + ',Bg1='
+                         + str(S_fit.column(1)[5]) + ',Bg2=' + str(S_fit.column(1)[6]) + ',constraints=('
+                         + str(gaussianFWHM) + '<GaussianFWHM),ties=(LorentzianFWHM=' + str(lorentzFWHM) + ')',
+                InputWorkspace=fnNoExt + '_monitor', MaxIterations=200, Output='T_fit')
+            #
+            DeleteWorkspace('T_fit_NormalisedCovarianceMatrix')
+            DeleteWorkspace(fnNoExt + '_monitor')
+            T_fit = mtd['T_fit_Parameters']
+            gaussian_FWHM_fitted = T_fit.column(1)[2]
+            width_T = np.sqrt(gaussian_FWHM_fitted ** 2 - gaussianFWHM_inst ** 2)
+            # Factor of 1e-3 converts back from meV to eV
+            Teff = (((width_T * 1e-3) ** 2) * e * ((1 + mass) ** 2)) / (4 * 1e-3 * T_fit.column(1)[0] * k * mass)
+            Teff_low = ((((width_T - T_fit.column(2)[2]) * 1e-3) ** 2) * e
+                        * ((1 + mass) ** 2))/(4 * 1e-3 * (T_fit.column(1)[0] + T_fit.column(2)[0]) * k * mass)
+            Teff_high = ((((width_T + T_fit.column(2)[2]) * 1e-3) ** 2) * e * ((1 + mass) ** 2)) /\
+                        (4 * 1e-3 * (T_fit.column(1)[0] - T_fit.column(2)[0]) * k * mass)
+            errTeff = 0.5*(Teff_high-Teff_low)
+            # ----------------------------------------------------------
+            # If the temperature is too far below the Debye temperature, then the result is inaccurate. Else the
+            # temperature is calculated assuming free gas formulation
+            # ----------------------------------------------------------
+            if 8 * Teff < 3 * TD:
+                self.log().information("The effective temperature is currently too far below the Debye temperature to"
+                                       "give an accurate measure.")
+                Tactual = Teff
+                Terror = errTeff
+            else:
+                Tactual = 3 * TD / (4 * np.log((8 * Teff + 3 * TD)/(8 * Teff - 3 * TD)))
+                Tactual_high = 3 * TD / (4 * np.log((8 * (Teff + errTeff) + 3 * TD) / (8 * (Teff + errTeff) - 3 * TD)))
+                Tactual_low = 3 * TD / (4 * np.log((8 * (Teff - errTeff) + 3 * TD) / (8 * (Teff - errTeff) - 3 * TD)))
+                Terror = 0.5 * (np.abs(Tactual - Tactual_high) + np.abs(Tactual - Tactual_low))
+            # Introduce a catchment for unphysically small determined errors, setting to defualt value if below a set
+            # threshold
+            Terror_flag = 0
+            if Terror < 5.0:
+                Terror_flag = 1
+                Terror = 10.0
+            if isDebug == 'True':
```

Comment:
I think this needs to be simply "if isDebug:" because the isDebug variable isn't a string